sdk_extensions/aws_v1_support/src/main/java/io/opentelemetry/sdk/extensions/trace/aws/resource/utils/DockerUtil.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.extensions.trace.aws.resource.utils;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+
+public class DockerUtil {
+
+  private static final Logger logger = Logger.getLogger(DockerUtil.class.getName());
+  private static final int CONTAINER_ID_LENGTH = 64;
+  private static final String DEFAULT_CGROUP_PATH = "/proc/self/cgroup";
+  private static final DockerUtil INSTANCE = new DockerUtil();
+
+  private DockerUtil() {}
+
+  public static DockerUtil getInstance() {
+    return INSTANCE;
+  }
+
+  /**
+   * Get docker container id from local cgroup file.
+   *
+   * @return docker container ID
+   */
+  @Nullable
+  @SuppressWarnings("DefaultCharset")
+  public String getContainerId() {
+    try (BufferedReader br = new BufferedReader(new FileReader(DEFAULT_CGROUP_PATH))) {
```

Comment:
Since this is one-time initialization logic, you may just stick with `Files.readAllLines`

Reply:
readAllLines is java 8+, so can't use that in this module, as-is